docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,400 @@
+successfully. It also contains statistics on the above-described timeout causes.
+
+Find more documentation on TAC reports in general [here](tac-reports).
+
+In the timeout case, the TAC reports contain some additional information that
+should help with diagnosing the timeout.
+
+
+### Finding timeout causes through modularization
+
+In addition to the other techniques described here, it can be insightful to
+remove parts of the code in order to isolate the timeout reason. If timeouts are
+eliminated through this, modular verification techniques can be employed in 
+order to prove correctness of the parts separately.
+These techniques are a relatively blunt instrument, but can be necessary in 
+particular with large or complex code bases.
+
+#### Sanity rules
+
+One way of isolating the timeout cause is by running with a trivial
+specification.  This way, the specification is ruled out as the source of
+complexity. Thus, a timeout on such a rule hints towards some parts of the
+program code being challenging for the solver, rather than the program code in
+combination with another, less trivial, spec.
+
+Sanity rules are such trivial specifications. For documentation on them, see
+{ref}`sanity <built-in-sanity>` and {ref}`deep sanity <built-in-deep-sanity>`. 
+
+(timeout-causes-library-contracts)=
+#### Library contracts
+
+Some systems are based on multiple library contracts which implement the
+business logic. They also forward storage updates to a single external contract
+holding the storage.
+
+In these systems, it can be appropriate to verify each library independently.
+
+If you encounter timeouts when trying to verify the main entry point contract to
+the system, check the impact of the libraries on the verification by summarizing
+all external library (delegate) calls as `NONDET`, using the option
+`-summarizeExtLibraryCallsAsNonDetPreLinking` as follows:
+
+```sh
+certoraRun ... --prover_args '-summarizeExtLibraryCallsAsNonDetPreLinking true'
+```
+
+```{note}
+This option is only applied for `delegatecall`s and _external_ library calls.
+Internal calls are automatically inlined by the Solidity compiler and are 
+subject to summarizations specified in the spec file's `methods` block.
+```
+
+(timeout-prevention)=
+# Timeout prevention
+
+Timeout prevention approaches fall into these categories.
+1. changing tool settings
+2. changing specs
+3. changing source code
+
+Changing tool settings is least invasive and easy to do, thus it is usually
+preferable to the other options. However, there are cases when parts of the
+input code that are very hard to reason about need to be worked around.
+Sometimes a combination of approaches is needed to resolve a timeout.
+
+
+In the following we will discuss some concrete approaches to timeout prevention.
+This collection will be extended over time based on user's experiences and tool
+improvements.
+
+```{note}
+The old documentation has a section on
```

Comment:
Referring to the old documentation is not ideal.

Reply:
Well, I consider it more of a "lesser evil", we should definitely update the old docs ... I figure in the meantime they have some value (?)